### ISSUE TITLE
[Request] Mark specific package as devel

### ISSUE THREAD

**Issue by nihaals** [Jguer/yay#1340]
**Is your feature request related to a problem? Please describe.**
Yay doesn't work very well with [`visual-studio-code-insiders` AUR](https://aur.archlinux.org/packages/visual-studio-code-insiders), it's essentially a devel package but without the `-git` suffix as it doesn't use a VCS. This means yay won't check for updates and you have to reinstall it to update (you don't know when there may be one so you would probably have to do a cronjob to reinstall it every day).

**Describe the solution you'd like**
Provide a way to mark a package as devel (or maybe another term) locally which means the `pkgver` function will be evaluated each time when checking for an update.

A flag that evaluates every package that has a `pkgver` function may also work but I'm not sure how wasteful this would be.

**Describe alternatives you've considered**
Cronjob that reinstalls all packages like this every 6 hours

**Comment by Jguer** [comment]
yay does not source, evaluate or run `PKGBUILD`s without your consent ( this could be a big security risk). Unfortunately besides this unsafe approach I do not see how this feature could be implement

**Comment by nihaals** [comment]
Could some kind of extension system help? Maybe you write a Go file or an executable which is not automatically updated (almost like Pacman Hooks) and this is used to almost rebind the pkgver function. So you could effectively say "instead of using this package's pkgver, use this executable instead" and then use that in the version check. I just started using Arch in the last week so I don't really know about standard approaches and safe ways of doing things so sorry if this wouldn't be doable, just it seems like the best way to install vsc insiders would be Snap

**Comment by grandchild** [comment]
tl;dr: Too much effort would be needed for too little gain.
- - - - - - - - - -
There are four options for various kinds of devel packages, that I can see:
* `pkgver` is run every check --> **Not good**, because there's no (generic) way to prevent a PKGBUILD from running `rm -rf ~` or other crap.
* The source repository of the upstream software is queried for a new version. --> This is what `--devel` currently does if I'm not mistaken.
* The Maintainer updates the pkgver in the PKGBUILD and thus updates it in AUR --> The maintainer of _visual-studio-code-insiders_ has already said they won't do that (or at least not "every day").
* An external user-specified and local pkgver-tool checks for the version of a package. --> Every such package would have to have a corresponding trusted pkgver script (because they might all be different), and someone would have to maintain these. It's an interesting proposal, and I can imagine it, but I cannot imagine it actually becoming reality, because the effort seems high in relation to the reward.

I think the best bet would be to convince the package maintainer to keep at least some rhythm of updating the PKGBUILD (weekly? monthly?).
Why? Because that's more in line with what *maintaining* a package actually means, namely pushing updates to the consumers of that package -- and so all the other pieces fall in line _much_ more easily: package managers just work, no new software has to be written, peoples' expectations and habits don't have to change etc.

**Comment by nihaals** [comment]
I don't think this needs to stay open since:
- I [made my own package for Visual Studio Code Insiders](https://github.com/nihaals/visual-studio-code-insiders-arch)
- I agree that the package maintainer should keep it updated
- This is the only package I've seen that has needed a feature like this (very few packages would benefit from this feature request)
